libraries/chain/asset_evaluator.cpp

```diff
@@ -328,6 +330,51 @@ void_result asset_update_evaluator::do_apply(const asset_update_operation& o)
    return void_result();
 } FC_CAPTURE_AND_RETHROW( (o) ) }
 
+void_result asset_update_issuer_evaluator::do_evaluate(const asset_update_issuer_operation& o)
+{ try {
+   database& d = db();
+
+   const asset_object& a = o.asset_to_update(d);
+   auto a_copy = a;
+   a_copy.validate();
+
+   FC_ASSERT(d.find_object(o.new_issuer));
+   if( a.is_market_issued() && o.new_issuer == GRAPHENE_COMMITTEE_ACCOUNT )
+   {
+      const asset_object& backing = a.bitasset_data(d).options.short_backing_asset(d);
+      if( backing.is_market_issued() )
+      {
+         const asset_object& backing_backing = backing.bitasset_data(d).options.short_backing_asset(d);
+         FC_ASSERT( backing_backing.get_id() == asset_id_type(),
+                    "May not create a blockchain-controlled market asset which is not backed by CORE.");
+      } else
+         FC_ASSERT( backing.get_id() == asset_id_type(),
+                    "May not create a blockchain-controlled market asset which is not backed by CORE.");
+   }
```

Comment:
For bonus points, refactor the parts that are identical with asset_update into a separate method.

Reply:
I am not sure what the best way to do this would be.
Do I simply setup a new function that tests `new_issuer`, or is some other construct better?

Reply:
Yes, I'd simply move the code block into a static function.

Reply:
I fear, this is too difficult for me and I need some guidance for C++. The major issue is that `o` is of type `asset_update_operation` or of type `asset_update_issuer_operation`. I am not sufficiently into templates and/or visitors to produce code that can do that. Could you give me a hint, please?

Reply:
KISS. ;-)
See latest commit.